core/shared/src/main/scala/coursier/util/Parse.scala

```diff
@@ -113,44 +113,115 @@ object Parse {
     }
   }
 
+  case class ModuleParseError(private val message: String = "",
```

Comment:
Exceptions shouldn't be case classes. Two `ModuleParseError` with the same `message` and `cause` would be considered equal (because case class), but may not actually be equal (thrown from different places, so different stack trace). But a simple class is ok.

Reply:
corrected.